src/RFC-1101_Tapplet.md

```diff
@@ -0,0 +1,123 @@
+Additionally the idea is to create an on-chain and association between smart contracts and their front end tapplets. This way, when a smart contract gets reused for a different purpose, it already comes with a ready tapplet that can be nested into other tapplets.
+
+### Tapplets
+
+At first, it is proposed that tapplets are npm packages published to the public npm registry, but in the future it may be extended also to other package managers. Npm is the first choice because it is widely used, so publishing a new Tapplet is as simple as publishing an npm package.
+
+Tapplets are separate packages and the file structure depends on its authors, however every package should contain a helpful readme file and must contain a manifest json file with Tapplet data.
+The following details are specific to tapplets:
+
+- The *version* string field in `package.json` and `tapplet.manifest.json` should match.
+- The *repository.url* field in `package.json` should match the correct repository for the Tapplet.
+- The *source.location.npm.packageName* in `tapplet.manifest.json` should match the name in `package.json`.
+- The *displayName* in `tapplet.manifest.json` should be a human-readable name
+- The image specified in *design.logoPath* in the manifest file is used as the logo displayed in tapplets market. This logo should be a valid SVG.
+
+After publishing the Tapplet, it is possible to connect to the Tapplet by using the Tapplet ID `npm:[packageName]`.
+
+### Manifest
+
+The Tapplet Manifest file specifies the most significant Tapplet’s data, which includes: package name, version, source and content hash. In the future manifest files may also contain references to other applets that get nested within.
+Before displaying the tapplet, its integrity will be verified against the hash stored in the Tapplet Registry.
+
+Proposed `tapplet.manifest.json` file:
+
+```
+{
+  "packageName": "@company-name/tapplet-name",
+  "version": "1.2.3",
+  "displayName": "Tapplet human readable name",
+  "status": "latest / verified / deprecated / vulnerable - anything like this",
+  "about": {
+    "summary": "Short (a few words) summary.",
+    "description": "Longer (a few sentences) project description."
+  },
+  "design": {
+    "logoPath": "./assets/logo.svg",
+    "backgroundPath": "./assets/background.svg"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/CompanyName/example-tapplet.git"
+  },
+  "source": {
+    "shasum": "shasumexample",
+    "location": {
+      "npm": {
+        "packageName": "@company-name/tapplet-name",
+        "registry": "https://registry.npmjs.org/"
+      }
+    }
+  },
+  "manifestVersion": "1.2.3"
+}
+
+```
+
```

Comment:
Some questions that arise that aren't clear from the current text:

* How do you propose managing upgrades? It's a complex topic on it's own to be sure.
* Where does the manifest live / Are there commitments to it anywhere?
* The manifest is mutable. How do you handle attempts to change the content hash (to point to a malicious app) without changing the version? (If it's meant to be immutable, the status field isn't useful.)

Reply:
These are really good points! I'll described them detailed in the RFC document, here just briefly. 
1. Each version of a tapplet needs to be registered separately. Once registered and listed in Tapplets Registry (ofc with checksum and so on) it is not planned to changed any data and upgrade is done within new PR to the Tapplet Registry. Only codeowner of a specific version can do it.
2. Tapplet's manifest, `tapplet.manifest.json`, is placed in the tapplet package (I'll add sample tapplet package structure to visualize it)
3. Before adding a tapplet to the Registry, CI will check if proposed package has correct structure, valid checksum etc. If does - next CI will generate `tappletsRegistry.manifest.json` file. So the contect hash can't be updated without checking an app before. Next version can be added with a new PR to the Tapplets Registry repo only by the codeowner.